packages/dht/src/dht/DhtNode.ts

```diff
@@ -0,0 +1,645 @@
+                    this.addNewContact(newContact.getPeerDescriptor())
+                    break
+                }
+            }
+        })
+        this.bucket.on('removed', (contact: DhtPeer) => {
+            //if (this.appId === DEFAULT_APP_ID) {
+            if (this.cleanUpHandleForConnectionManager) {
+                const connectionManager = this.cleanUpHandleForConnectionManager!
+                connectionManager.disconnect(contact.getPeerDescriptor())
+            }
+            logger.trace(`Removed contact ${contact.peerId.value.toString()}`)
+            this.emit(Event.CONTACT_REMOVED, contact.getPeerDescriptor())
+        })
+        this.bucket.on('added', async (contact: DhtPeer) => {
+            if ( !contact.peerId.equals(this.ownPeerId!) ) {
+                if (await contact.ping(this.ownPeerDescriptor!)) {
+                    logger.trace(`Added new contact ${contact.peerId.value.toString()}`)
+                    this.emit(Event.NEW_CONTACT, contact.getPeerDescriptor())
+                } else {
+                    this.removeContact(contact.getPeerDescriptor())
+                    this.addClosestContactToBucket()
+                }
+            }
+        })
+        this.bucket.on('updated', (_oldContact: DhtPeer, _newContact: DhtPeer) => {
+            // TODO: Update contact info to the connection manager and reconnect
+        })
+        this.neighborList = new SortedContactList(selfId, this.MAX_NEIGHBOR_LIST_SIZE)
+        this.openInternetPeers = new SortedContactList(selfId, this.MAX_NEIGHBOR_LIST_SIZE / 2)
+    }
+
+    public getNeighborList(): SortedContactList {
+        return this.neighborList!
+    }
+
+    public getNodeId(): PeerID {
+        return this.ownPeerId!
+    }
+
+    public onGetClosestPeers(caller: PeerDescriptor): DhtPeer[] {
+        if (!this.started || this.stopped) {
+            return []
+        }
+        logger.trace(`processing getClosestPeersRequest`)
+        const ret = this.bucket!.closest(caller.peerId, 5)
+        this.addNewContact(caller, true)
+        //this.neighborList!.setContacted(PeerID.fromValue(caller.peerId))
+        return ret
+    }
+
+    public async onRoutedMessage(routedMessage: RouteMessageWrapper): Promise<void> {
+        if (!this.started || this.stopped || this.routerDuplicateDetector.test(routedMessage.nonce)) {
+            return
+        }
+        logger.trace(`Processing received routeMessage ${routedMessage.nonce}`)
+        this.addNewContact(routedMessage.sourcePeer!, true)
+        this.routerDuplicateDetector.add(routedMessage.nonce)
+        const message = Message.fromBinary(routedMessage.message)
+        if (this.ownPeerId!.equals(PeerID.fromValue(routedMessage.destinationPeer!.peerId))) {
+            logger.trace(`RouteMessage ${routedMessage.nonce} successfully arrived to destination`)
+            this.emit(ITransportEvent.DATA, routedMessage.sourcePeer, message)
```

Comment:
```suggestion
        if (this.ownPeerId!.equals(PeerID.fromValue(routedMessage.destinationPeer!.peerId))) {
            logger.trace(`RouteMessage ${routedMessage.nonce} successfully arrived to destination`)
            this.emit(ITransportEvent.DATA, routedMessage.sourcePeer, Message.fromBinary(routedMessage.message))
```

Would be clear that the parsed message is used only in the first `if` branch

Reply:
The appId of the routed message is fetched from the parsed message so this change cannot be made. The appId of the routed message must remain the same between routing nodes as routing messages to ie. Layer2 would be impossible without it.